src/consensus/vote_transaction.zig

```diff
@@ -0,0 +1,692 @@
+        return switch (self.*) {
+            .vote => |vote| vote.slots.len,
+            .vote_state_update, .compact_vote_state_update => |vote_state_update| vote_state_update
+                .lockouts.items.len,
+            .tower_sync => |tower_sync| tower_sync.lockouts.items.len,
+        };
+    }
+
+    pub fn hash(self: *const VoteTransaction) Hash {
+        return switch (self.*) {
+            .vote => |vote| vote.hash,
+            .vote_state_update, .compact_vote_state_update => |vote_state_update| vote_state_update
+                .hash,
+            .tower_sync => |tower_sync| tower_sync.hash,
+        };
+    }
+
+    pub fn eql(self: *const VoteTransaction, other: *const VoteTransaction) bool {
+        if (@intFromEnum(self.*) != @intFromEnum(other.*)) {
+            return false;
+        }
+
+        return switch (self.*) {
+            .vote => |self_vote| {
+                const other_vote = other.vote;
+                return std.mem.eql(Slot, self_vote.slots, other_vote.slots) and
+                    self_vote.hash.eql(other_vote.hash) and
+                    self_vote.timestamp == other_vote.timestamp;
+            },
+            .vote_state_update => |self_vsu| {
+                const other_vsu = other.vote_state_update;
+                if (self_vsu.lockouts.items.len != other_vsu.lockouts.items.len or
+                    !self_vsu.hash.eql(other_vsu.hash) or
+                    self_vsu.timestamp != other_vsu.timestamp)
+                {
+                    return false;
+                }
+                for (
+                    self_vsu.lockouts.items,
+                    other_vsu.lockouts.items,
+                ) |self_lockout, other_lockout| {
+                    if (self_lockout.slot != other_lockout.slot or
+                        self_lockout.confirmation_count != other_lockout.confirmation_count)
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            },
+            .compact_vote_state_update => |self_vsu| {
+                const other_vsu = other.compact_vote_state_update;
+                if (self_vsu.lockouts.items.len != other_vsu.lockouts.items.len or
+                    !self_vsu.hash.eql(other_vsu.hash) or
+                    self_vsu.timestamp != other_vsu.timestamp)
+                {
+                    return false;
+                }
+                for (
+                    self_vsu.lockouts.items,
+                    other_vsu.lockouts.items,
+                ) |self_lockout, other_lockout| {
+                    if (self_lockout.slot != other_lockout.slot or
+                        self_lockout.confirmation_count != other_lockout.confirmation_count)
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            },
+            .tower_sync => |self_ts| {
```

Comment:
Correct me if I'm wrong, but aren't the branches for `tower_sync`, `compact_vote_state_update`, and `vote_state_update` all identical here?
If yes, can we unify them? Can use an inline prong:

```zig
inline //
.vote_state_update,
.compact_vote_state_update,
.tower_sync,
=> |self_payload, tag| {
    const other_payload = @field(other, @tagName(tag));
    // -- snip --
},
```

Reply:
[8c36719](https://github.com/Syndica/sig/pull/664/commits/8c36719826a7938ce7a19c117df89c7b3c9905fc)